R/test-threads.R

```diff
@@ -0,0 +1,268 @@
+#' @param .tags a character vector identifying the tags of the models you want to delete.
+#'              If set to `NULL`, no filtering will be performed and all associated model files will be deleted.
+#' @param .force logical (T/F). If `TRUE`, do not prompt the user if they want to delete the models.
+#'
+#' @importFrom checkmate check_character
+#' @importFrom tidyr crossing
+#' @importFrom dplyr arrange
+#' @importFrom utils askYesNo
+#'
+#' @export
+cleanup_mods <- function(.mods, .tags = "test threads", .force = FALSE){
+  if(!is.null(.tags)) check_character(.tags)
+
+  if(!inherits(.mods, "bbi_nonmem_model")){
+    check_model_object_list(.mods)
+  }else{
+    check_model_object(.mods)
+    .mods <- list(.mods)
+  }
+
+  mod_info <- map_dfr(.mods, function(mod.x){
+    mod_tags <- mod.x$tags
+    mod_tags <- paste(mod_tags, collapse = ", ")
+    mod_tags <- ifelse(mod_tags == "", "NA", mod_tags)
+    tibble::tibble(
+      mod_paths = mod.x$absolute_model_path,
+      mod_thread = mod.x$bbi_args$threads,
+      mod_tags = mod_tags
+    )
+  })
+
+  tag_groups <- if(is.null(.tags)){
+    crossing(mod_tags = mod_info$mod_tags, .tags = "NA", found = TRUE) %>% left_join(mod_info, by = "mod_tags")
+  }else{
+    tag_levels <- unique(.tags) # message in order of tags
+    tag_groups <- crossing(mod_tags = mod_info$mod_tags, .tags) %>% left_join(mod_info, by = "mod_tags") %>%
+      mutate(.tags = ordered(.tags, levels = tag_levels)) %>% arrange(.tags)
+    found <- map2(tag_groups$mod_tags, tag_groups$.tags, function(tag.x, .tag){
+      grepl(.tag, tag.x)
+    }) %>% unlist()
+
+    tag_groups$found <- found
+
+    if(any(tag_groups$found==FALSE)){
+      not_found <- c()
+      for(tag.x in .tags){
+        tag.found <- tag_groups$found[tag_groups$.tags==tag.x]
+        if(all(tag.found==FALSE)){
+          not_found <- c(not_found, tag.x)
+        }
+      }
+      if(length(not_found) > 0){
+        message("The following tags were not found:\n",
+                paste("-",unique(not_found),"\n"))
+      }
+    }
+    tag_groups[found,]
+  }
+
+  mod_paths <- unique(tag_groups$mod_paths)
+
+  if(length(mod_paths)==0){
+    stop("None of specified tags were found")
+  }
+
+  mods_removed <- unique(tag_groups$mod_tags)
+
+  msg_remove <- paste0(
+    paste("Removed", length(mod_paths), "models with the following tags:\n"),
+    paste("-",mods_removed, collapse = "\n")
+  )
+
+  if(.force){
+    for (m in mod_paths) {
+      if (fs::file_exists(yaml_ext(m))) fs::file_delete(yaml_ext(m))
+      if (fs::file_exists(ctl_ext(m))) fs::file_delete(ctl_ext(m))
+      if (fs::dir_exists(m)) fs::dir_delete(m)
+    }
+    message(msg_remove)
+  }else{
+    msg_prompt <- paste0(
+      paste("Are you sure you want to remove", length(mod_paths), "models with the following tags?: "),
+      paste0("`",mods_removed,"`", collapse = ", ")
+    )
+    delete_prompt <- askYesNo(msg_prompt)
+    if(delete_prompt){
```

Comment:
this needs to `if (isTRUE(delete_prompt))` because otherwise answering "cancel" throws this:

```
Error in if (delete_prompt) { : missing value where TRUE/FALSE needed
```